index.js

```diff
@@ -75,53 +81,60 @@ async function main() {
 
     // Install packages
     if (core.getInput("packages")) {
-      var pkgSpecs = core.getInput("packages").split(",");
-      const repositories = ["https://packages.opentap.io"];
-
-      const repository = core.getInput("additional-repository");
-      const hasRepository = !!repository;
-      if (hasRepository) {
-        repositories.push(repository);
-
-        const token = core.getInput("repository-token");
-        const hasToken = !!token;
-        // If a version was specified, verify it is recent enough to support user tokens
-        if (hasVersion && hasToken) {
-          const majorMinorPattern = /^(?<major>\d+)\.(?<minor>\d+).*/;
-          const match = majorMinorPattern.exec(version);
-          const minor = Number(match.groups["minor"]);
-          if (minor <= 21) {
-            core.setFailed(
-              "repository-token support requires OpenTAP 9.22.0 or greater."
-            );
-            return;
-          }
+      const domainPattern = /^(?<scheme>https?:\/\/)?(?<domain>.+)$/;
+
+      const repositories = [
+        {
+          url: "https://packages.opentap.io",
+          domain: domainPattern.exec("https://packages.opentap.io").groups["domain"],
+          token: core.getInput("token")
         }
+      ];
+      const additionalRepository = core.getInput("additional-repository");
+      if (!!additionalRepository){
+        repositories.push({
+          url: additionalRepository,
+          domain: domainPattern.exec(additionalRepository).groups["domain"],
+          token: core.getInput("additional-repository-token")
+        });
+      }
 
-        // If a token was specified, it should be written to AuthenticationSettings.xml in the tap installation.
-        if (hasToken) {
-          // Optionally match a scheme and a domain.
-          const domainPattern = /^(?<scheme>https?:\/\/)?(?<domain>.+)$/;
-          const match = domainPattern.exec(repository);
-          const domain = match.groups["domain"];
-
-          var authenticationSettings = GetAuthenticationSettings(domain, token);
-          const settingsDir = destDir + "/Settings/";
-          const destFile = settingsDir + "AuthenticationSettings.xml";
-          if (!fs.existsSync(settingsDir)) {
-            fs.mkdirSync(settingsDir);
-          }
-          fs.writeFileSync(destFile, authenticationSettings);
-          core.debug(`Wrote authentication settings to ${destFile}`);
+      const hasToken = repositories.some(r => !!r.token);
+      // If an OpenTAP version was specified, verify it is recent enough to support user tokens
+      if (hasVersion && hasToken) {
+        const majorMinorPattern = /^(?<major>\d+)\.(?<minor>\d+).*/;
+        const match = majorMinorPattern.exec(version);
+        const minor = Number(match.groups["minor"]);
+        if (minor <= 21) {
```

Comment:
OpenTAP 9.21.1 supports user tokens: https://github.com/opentap/opentap/blob/v9.21.1/Engine/Authentication/TokenInfo.cs

Can we change the check to 9.21.1 ?